examples/src/bin/autocomplete.rs

```diff
@@ -0,0 +1,95 @@
+use cursive::align::HAlign;
+use cursive::traits::Scrollable;
+use cursive::view::{Boxable, Identifiable};
+use cursive::views::{Dialog, EditView, LinearLayout, SelectView, TextView};
+use cursive::Cursive;
+use lazy_static::lazy_static;
+
+// This example shows a way to implement a (Google-like) autocomplete search box.
+// Try entering "tok"!
+
+lazy_static! {
+    static ref CITIES: &'static str = include_str!("../../assets/cities.txt");
+}
+
+fn main() {
+    let mut siv = cursive::default();
+
+    siv.add_layer(
+        Dialog::around(
+            LinearLayout::vertical()
+                // the query box is on the top
+                .child(
+                    EditView::new()
+                        // update results every time the query changes
+                        .on_edit(on_edit)
+                        // submit the focused (first) item of the matches
+                        .on_submit(on_submit)
+                        .with_name("query"),
+                )
+                // search results below the input
+                .child(
+                    SelectView::new()
+                        // shows all cities by default
+                        .with_all_str(CITIES.lines())
+                        // Sets the callback for when "Enter" is pressed.
+                        .on_submit(show_next_window)
+                        // Center the text horizontally
+                        .h_align(HAlign::Center)
+                        .with_name("matches")
+                        .scrollable()
+                        .fixed_size((20, 10)),
```

Comment:
Do we actually care about the width? We could probably just `fixed_height(10)`?

Reply:
I borrowed these two lines from the other example, `select.rs`, which uses the same set of data. You are right, in this case, the width of the query may exceed 20, so it would be better not to fix the width.